presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/aggregation/TestUnionAgg.java

```diff
@@ -0,0 +1,399 @@
+                {
+                        "square with adjacent point becomes geometry collection",
+                        "GEOMETRYCOLLECTION (POINT (5 2), POLYGON ((1 1, 3 1, 3 3, 1 3, 1 1)))",
+                        new String[] {"POLYGON ((1 1, 3 1, 3 3, 1 3, 1 1))", "POINT (5 2)"},
+                },
+        };
+    }
+
+    private InternalAggregationFunction function;
+
+    @BeforeClass
+    public void registerFunctions()
+    {
+        GeoPlugin plugin = new GeoPlugin();
+        for (Type type : plugin.getTypes()) {
+            functionAssertions.getTypeRegistry().addType(type);
+        }
+        functionAssertions.getMetadata().addFunctions(extractFunctions(plugin.getFunctions()));
+
+        function = functionAssertions.getMetadata().getFunctionRegistry().getAggregateFunctionImplementation(new Signature(
+                "union_agg",
+                FunctionKind.AGGREGATE,
+                TypeSignature.parseTypeSignature(GeometryType.GEOMETRY_TYPE_NAME),
+                TypeSignature.parseTypeSignature(GeometryType.GEOMETRY_TYPE_NAME)));
+    }
+
+    @Test(dataProvider = "point")
+    public void testPoint(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "linestring")
+    public void testLineString(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "polygon")
+    public void testPolygon(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "multipoint")
+    public void testMultiPoint(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "multilinestring")
+    public void testMultiLineString(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "multipolygon")
+    public void testMultiPolygon(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    @Test(dataProvider = "geometrycollection")
+    public void testGeometryCollection(String testDescription, String expectedWkt, String... wkts)
+    {
+        assertAggregatedGeometries(testDescription, expectedWkt, wkts);
+    }
+
+    private void assertAggregatedGeometries(String testDescription, String expectedWkt, String... wkts)
+    {
+        List<Slice> geometrySlices = Arrays.stream(wkts)
+                .map(text -> text == null ? null : OGCGeometry.fromText(text))
+                .map(input -> input != null ? GeometrySerde.serialize(input) : null)
```

Comment:
`input == null ? null : GeometrySerde.serialize(input)` for consistency with the line above